protocol/messenger.go

```diff
@@ -1337,6 +1338,50 @@ func (m *Messenger) watchExpiredMessages() {
 	}()
 }
 
+// watchUnmutedChats regularly checks for chats that should be unmuted
+func (m *Messenger) watchUnmutedChats() {
+	m.logger.Debug("watching unmuted chats")
+	go func() {
+		for {
+			chats, _ := m.persistence.Chats()
+			select {
+			case <-time.After(3 * time.Second): // Poll every 3 seconds
+				for _, chat := range chats {
+					switch chat.MuteType {
+					case MuteFor15MinsType:
+						if time.Now().Unix()-chat.MutedAt > MuteFor15MinsDuration && chat.MuteType != 0 {
+							_ = m.persistence.UnmuteChat(chat.ID)
+							m.config.messengerSignalsHandler.ChatUnmutedEvent(chat.ID)
```

Comment:
Here I would just get all the chats and signal once with a `MessengerResponse`, that way you don't have to handle a new signal in the frontend